src/SideBar.tsx

```diff
@@ -31,11 +32,24 @@ export const SideBar = ({
   isDesktopSideBarOpen,
 }: SideBarProps) => {
   const [isSearchHistoryOpen, setIsSearchHistoryOpen] = useState(false);
+  const [searchableHistory, setSearchableHistory] =
+    useState<SearchableChatHistories | null>(null);
+
+  const onClickSearchHistory = async () => {
+    try {
+      const history = await getSearchableHistory();
+      setSearchableHistory(history);
+      setIsSearchHistoryOpen(true);
+    } catch (error) {
+      console.error('Failed to load search history:', error);
+    }
+  };
```

Comment:
We only need this specific data when the search modal is open, so let's initialize it as `null` and then on successful fetch, update the value.